tests/test_restart.py

```diff
@@ -0,0 +1,445 @@
+                           run_full_end_date.strftime('%Y%m%d')),
+            'states_{}_{}'.format(run_full_end_date.strftime('%Y%m%d'),
+                                  statesec))
+        if state_format == 'ASCII':
+            states_full_run = read_ascii_state(state_fname)
+        elif state_format == 'BINARY':
+            states_full_run = read_binary_state(state_fname)
+    elif driver == 'image':
+        state_fname = os.path.join(
+            state_basedir,
+            '{}_{}'.format(run_full_start_date.strftime('%Y%m%d'),
+                           run_full_end_date.strftime('%Y%m%d')),
+            'states.{}_{}.nc'.format(run_full_end_date.strftime('%Y%m%d'),
+                                     statesec))
+        ds_states_full_run = xr.open_dataset(state_fname)
+
+    # --- Read the state at the end of the last period of run --- #
+    # Extract the last split run period
+    run_last_period_start_date = run_periods[-1]['start_date']
+    run_last_period_end_date = run_periods[-1]['end_date']
+    # Read the state file
+    if driver == 'classic':
+        state_fname = os.path.join(
+            state_basedir,
+            '{}_{}'.format(run_last_period_start_date.strftime('%Y%m%d'),
+                           run_last_period_end_date.strftime('%Y%m%d')),
+            'states_{}_{}'.format(run_last_period_end_date.strftime('%Y%m%d'),
+                                  statesec))
+        if state_format == 'ASCII':
+            states = read_ascii_state(state_fname)
+        elif state_format == 'BINARY':
+            states = read_binary_state(state_fname)
+    elif driver == 'image':
+        state_fname = os.path.join(
+            state_basedir,
+            '{}_{}'.format(run_last_period_start_date.strftime('%Y%m%d'),
+                           run_last_period_end_date.strftime('%Y%m%d')),
+            'states.{}_{}.nc'.format(
+                    run_last_period_end_date.strftime('%Y%m%d'),
+                    statesec))
+        ds_states = xr.open_dataset(state_fname)
+
+    # --- Compare split run states with full run --- #
+    if driver == 'classic':
+        # If ASCII state file, check if almost the same
+        if state_format == 'ASCII':
+            np.testing.assert_almost_equal(states, states_full_run, decimal=3)
+        # If BINARY state file, check if exactly the same
+        elif state_format == 'BINARY':
+            if states != states_full_run:
+                raise VICTestError('Restart causes inexact state outputs!')
+            else:
+                return
+    elif driver == 'image':
+        for var in ds_states.data_vars:
+            if np.absolute(ds_states[var] - ds_states_full_run[var]).\
+               max().values > 0.0:
+                raise VICTestError('Restart causes inexact state outputs'
+                                   '(var {}) !'.format(var))
+            else:
+                continue
+        return
```

Comment:
since you don't share any of the test code here between the classic and image drivers below line 360, just do a single if/elif block.

also, the `else: continue` blocks are not needed in here.